### DISCUSSION TITLE
Integrate AI / GPT-4 to make docs assistant

### DISCUSSION THREAD

**Discussion by ntucker** [facebook/docusaurus#8835] [Ideas]
https://openai.com/customer-stories/stripe

Stripe has already done this with their docs. This would really improve people's ability to explore documentation

**Comment by slorber** [comment]
I don't know much about how these things work myself right now but this can probably be implemented as a community plugin if someone is willing to build it. I probably won't have time to work on it myself immediately but I'd be there to support in case of any blocker.

Afaik @MaxSavonin is interested to build such a product for Docusaurus

@michaelfester built a useful service that could be probably be used with Docusaurus: https://markprompt.com/

**Reply by remcohaszing** [reply]
Just giving a heads up I'm working on Markprompt integration. 🙂

**Reply by windhamdavid** [reply]
I had no issues just adding the Markprompt npm package and using the react component in an .mdx file.

**Reply by schneegansm** [reply]
@remcohaszing did the Markprompt integration.🙂 Here are the docs: https://markprompt.com/docs#docusaurus

**Reply by slorber** [reply]
Related blog post to look at: [Markprompt + Algolia: combining prompts with instant search for docs](https://markprompt.com/blog/algolia)

**Comment by jhcao23** [comment]
I believe there will be a central place to intelligently search all good open-source project docs, otherwise Algolia might be willing to work hard toward it :)

**Comment by slorber** [comment]
Another cool solution ([Asktro](https://asktro.com/)) presented here with [Docusaurus integration docs](https://asktro.com/docs/docusaurus) available: 

https://github.com/facebook/docusaurus/discussions/9198

**Reply by skovy** [reply]
Thanks for posting on this discussion! I would appreciate feedback from folks if a solution like this works well, or if they are looking for something different. 

I think there's an opportunity to make this open source in some way and "bring your own OpenAI key." The other challenge is having a vector store, but maybe using something like [Flexsearch](https://github.com/nextapps-de/flexsearch) (Nextra uses this by default) instead of a vector store + bringing your own OpenAI API key would be the most open source and setup friendly?

**Reply by samducker** [reply]
Good idea for a feature, like the open source ish idea.

**Reply by alex-zwingli** [reply]
Please remove this link. Whether it's been broken or not, it leads to a malicious site now.

**Comment by slorber** [comment]
Here's a PR from @nickscamara proposing to integrate [Mendable AI](https://www.mendable.ai/) to the Jest Docusaurus site: 
https://github.com/jestjs/jest/pull/14471

![image](https://github.com/facebook/docusaurus/assets/749374/8319a19e-d247-4e3a-a435-0b5fc32c2e08)

**Comment by fokkedekker** [comment]
I have built this here: https://github.com/seaplane-io/seaplane-demos/tree/main/docusaurus-chatbot

It's running on our Docuaurus pages: https://developers.seaplane.io/docs/seaplane-gpt

I am currently using streamlit but have been thinking about fully integrating it. The backend runs on our own platform (seaplane.io). Its using OpenAI as the LLM of choice but we support other LLMs too. 

Anyway would love to see if we can integrate this with Docusaurus, happy to help!

**Comment by dgarcia360** [comment]
[Biel.ai](https://biel.ai/) offers an AI chatbot + search solution that integrates seamlessly with Docusaurus.  Check out the integration guide—it only takes a few minutes to set up: [AI Chatbot for Docusaurus](https://docs.biel.ai/installation/docusaurus).

![image](https://github.com/user-attachments/assets/b02ede6a-0eb6-465c-ad3d-ad9b61749ea3)


**Disclaimer:** I’m the founder of Biel.ai! You can sign up for a free trial and configure the chatbot yourself or [contact us](https://biel.ai/contact) for a custom demo using your content.

**Comment by ahelmy** [comment]
Maybe it is late, but I developed a tool to add the AI assistant to any docusaurus.
https://github.com/ahelmy/docusaurus-ai
I will like to hear from anyone his thoughts.

**Comment by qtangs** [comment]
This might be worth a try: https://documate.site/integration/docusaurus
